qiskit_nature/second_q/problems/electronic_structure_problem.py

```diff
@@ -77,54 +76,47 @@ def filter_criterion_custom(eigenstate, eigenvalue, aux_values):
 
     """
 
-    def __init__(
-        self,
-        driver: ElectronicStructureDriver,
-        transformers: Optional[List[BaseTransformer]] = None,
-    ):
+    def __init__(self, hamiltonian: Hamiltonian) -> None:
         """
-
         Args:
-            driver: A fermionic driver encoding the molecule information.
-            transformers: A list of transformations to be applied to the driver result.
+            hamiltonian: the Hamiltonian of this problem. This needs to be an
+                :class:`.ElectronicEnergy`.
+
+        Raises:
+            TypeError: if the provided ``hamiltonian`` is not of type :class:`.ElectronicEnergy`.
         """
-        super().__init__(driver, transformers, "ElectronicEnergy")
+        if not isinstance(hamiltonian, ElectronicEnergy):
+            raise TypeError(
+                "Only an ElectronicEnergy hamiltonian is supported by the ElectronicStructureProblem,"
+                f" not one of type, {type(hamiltonian)}"
+            )
+        super().__init__(hamiltonian)
+        self.properties: ElectronicPropertiesContainer = ElectronicPropertiesContainer()
+        self.molecule: MoleculeInfo = None
+        self.basis_transform: ElectronicBasisTransform = None
```

Comment:
If we are going over to public instance vars instead of setter/getters thats fine. But they need to come out in docs and it can be done with Attrributes e.g see the new evolvers/PVQD that was added to Terra algorithms.

Reply:
The `basis_transform` is here only until the next refactoring with the `PolynomialTensor` can be done. Once that happens, the problem will have a single `basis` associated with it, and the `basis_transform` will no longer be stored inside of it. Instead, a `BasisTransformer` can be extracted which can transform the basis of an `ElectronicStructureProblem`